### PR TITLE
fix(Tile): fixes transitions on metadata metadata

### PR DESCRIPTION
**Pull request by arwehrman**
## Description

This PR does the following:

- fixes metadataPatch so bottom animation works  every time Tile goes from unfocus to focus
- restructures the code, breaking down components into sections for readability
- adds more logic to the shouldShowMetadata to account for all possible scenarios instead of just persistentMetadata and isFocused

## References

LUI-658

## Testing

- pull down branch
- `yarn start`
-  go to Tile story
- select `inset` in controls
- click `focus` in controls, bottom metadata data should move from bottom to top
- click `unfocus` and `focus` again the transition should perform as expected

## Checklist

- [ ] all commented code has been removed
- [ ] any new console issues have been resolved
- [ ] code linter and formatter has been run
- [ ] test coverage meets repo requirements
- [ ] PR name matches the expected semantic-commit syntax

### CODE REVIEW

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.js [hunk 1]
```diff
@@ -250,26 +176,24 @@ export default class Tile extends Surface {
...
     this._Background.alpha = 0; // Since the image is loaded the surface does not need to be shown
   }
 
+  /* ------------------------------ Badge ------------------------------ */
   _updateBadge() {
-    // Remove Badge if no longer required
-    if (!this.badge?.title || this._isCircleLayout) {
-      if (this._Badge) {
-        this._Content.patch({
-          Badge: undefined
```
Comment (erautenberg) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.js]: Why was this removed? I noticed that the Label equivalent is still there

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.js [hunk 2]
```diff
@@ -326,14 +252,37 @@ export default class Tile extends Surface {
...
+  get _badgeLabelTransitions() {
+    return {
+      y: [
+        this._shouldShowBadgeLabel ? this.style.paddingY : 0,
+        this.__shouldShowBadgeLabel
+          ? this.style.animationEntrance
+          : this.style.animationExit
+      ],
+      alpha: [
+        this._shouldShowBadgeLabel && !this._isCircleLayout ? 1 : 0.001,
```
Comment (erautenberg) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.js]: I'm a little confused by all the alpha logic for Badge and Label. You have this here, but also basically the same thing in the patch for the updateBadge and updateLabel methods.

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.js [hunk 3]
```diff
@@ -394,7 +347,7 @@ export default class Tile extends Surface {
       }
       return;
     }
-
+    // --------------------------------------//
```
Comment (erautenberg) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.js]: not sure what this bar is adding or signifying?

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.js [hunk 4]
```diff
@@ -441,11 +394,49 @@ export default class Tile extends Surface {
...
+  /* ------------------------------ Metadata  ------------------------------ */
+
+  get _shouldShowMetadata() {
+    return this._persistentMetadata || this._isFocusedMode;
+  }
+
+  get _isInsetMetadata() {
+    return this._metadataLocation === 'inset';
+  }
+  // called in animateMetadata
```
Comment (erautenberg) [inline packages/@lightningjs/ui-components/src/components/Tile/Tile.js]: can this comment get moved inside the method or something? At a minimum, let's add an empty line after line 409.

**Review by erautenberg — CHANGES_REQUESTED** [review]
The Badge seems to be broken
Screen Shot 2023-05-16 at 10 11 19 AM

Also, let's check with Sean on the animation in and out. I think when going from focus to unfocus, the inset is supposed to slide back down (opposite of the slide up it does on the way in). Also it doesn't seem like it is alpha-ing in correctly, it appears at the bottom, then slides up, but doesn't fade in as it slides up.

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.js (hunk 3 above)
Comment (arwehrman) [reply packages/@lightningjs/ui-components/src/components/Tile/Tile.js]: left over comment from when I was troubleshooting something. Will remove

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.js (hunk 4 above)
Comment (arwehrman) [reply packages/@lightningjs/ui-components/src/components/Tile/Tile.js]: I'll remove the comment. It's a leftover from my notes

File: packages/@lightningjs/ui-components/src/components/Tile/Tile.js (hunk 1 above)
Comment (arwehrman) [reply packages/@lightningjs/ui-components/src/components/Tile/Tile.js]: I removed it when I was figuring out whether we patch the component in as `undefined` or alpha out the component

Comment (arwehrman) [reply packages/@lightningjs/ui-components/src/components/Tile/Tile.js]: I can add it back

**Review by erautenberg — APPROVED** [review]

**Review by ImCoolNowRight — APPROVED** [review]
